consensus/istanbul/validator/weighted.go

```diff
@@ -844,13 +845,17 @@ func calcWeight(weightedValidators []*weightedValidator, stakingAmounts []float6
 	localLogger.Debug("calculation weight finished")
 }
 
-func (valSet *weightedCouncil) refreshProposers(seed int64, blockNum uint64) {
+func (valSet *weightedCouncil) refreshProposers(seed int64, blockNum uint64, rules params.Rules) {
 	var candidateValsIdx []int // This is a slice which stores index of validator. it is used for shuffling
 
 	for index, val := range valSet.validators {
-		weight := val.Weight()
-		for i := uint64(0); i < weight; i++ {
+		if rules.IsKore {
```

Comment:
Is this newly added branch required?
I think when all weights are 1, `candidateValsIdx` will have the same output regardless of whether or not this branch is taken.

Reply:
If we remove the branch, `TestWeightedCouncil_RefreshWithNonZeroWeight_AfterKoreHardFork` will fail. However, I'm not sure if testing with `testNonZeroWeights` is fair because I don't think such an array of weights can be produced after Kore.

Reply:
That's what i want to ask.
I wanted to prevent wrong calculation from wrong weight.
Do you think it is ok to remove the branch and change the tests.

Reply:
ian0371 and me were taking about it.
It's no problem we thought.
I will change it.

Reply:
Reflected in 9f56fdb00e15c7824559616978fc2a410ee652f9.
Thank you!